apps/core/scenes/passengerForm/BaggageBundles/BaggageBundle.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+
+import React from 'react';
+import { View } from 'react-native';
+import { StyleSheet, RadioButton } from '@kiwicom/universal-components';
+import { Text, Separator } from '@kiwicom/margarita-components';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+
+import { type BagTitleType } from './BagTypes';
+import BaggageTypeDetail from './BaggageTypeDetail';
+
+type Bag = {|
+  +type: BagTitleType,
+  +quantity: number,
+  +dimensions: string,
+  +weight: string,
+|};
+
+type Props = {|
+  +price: ?string,
+  +bags: Array<Bag>,
+  +bagIndex: number,
+  +onToggleCheck: (BagTitleType, number) => void,
+  +type: string,
+  +currentlyChosen: number,
+|};
+
+export default function BaggageBundle({
+  currentlyChosen,
+  type,
+  bagIndex,
+  bags,
+  price,
+  onToggleCheck,
+}: Props) {
+  const shouldBeChecked = currentlyChosen === bagIndex;
+  function handleToggleCheck() {
+    onToggleCheck(type, bagIndex);
+  }
+
+  return (
+    <View
+      style={[styles.container, shouldBeChecked && styles.containerChecked]}
+    >
+      <View style={styles.row}>
+        <RadioButton
```

Comment:
It will be nice to have whole component clickable. We can use `label` prop and remove unnecessary View wraps. So it can look something like:

```
return (
    <RadioButton
      type="check"
      checked={shouldBeChecked}
      onPress={handleToggleCheck}
      style={[styles.container, shouldBeChecked && styles.containerChecked]}
      label={
        <View style={styles.baggageDetails}>
          <View style={styles.something}>
            {bags.map((bag, index) => {
              return (
                <BaggageTypeDetail
                  key={index}
                  type={bag.type}
                  quantity={bag.quantity}
                  dimensions={bag.dimensions}
                  weight={bag.weight}
                  chosen={shouldBeChecked}
                />
              );
            })}
            <Separator />
          </View>
          <Text>{price}</Text>
        </View>
      }
    />
  );
```

Reply:
When I tried it like that its complaining about `View` nested in the `Text` somehow

Reply:
True, sorry I just don't remember how component I've written actually works 🙂 .
It should be passed as children instead of label. But it's not that crucial for now so I also added it into TODO.